framework/validators/FilterValidator.php

```diff
@@ -78,7 +78,7 @@ public function validateAttribute($model, $attribute)
     {
         $value = $model->$attribute;
         if (!$this->skipOnArray || !is_array($value)) {
-            $model->$attribute = call_user_func($this->filter, $value);
+            $model->$attribute = call_user_func((string)$this->filter, $value);
```

Comment:
I'm pretty sure that passing empty string (result of `(string) null`) a first argument to `call_user_func()` is also incorrect.

Reply:
What would you suggest?

Reply:
#19533 is clearly about scenario when `filter` contains `'trim'`. Solution would be to use `trim` validator in Gii instead `'filter', 'filter' => 'trim'`. `trim` validator was already adjusted for PHP 8.1.

Reply:
I believe the actual Deprecated error is called on the passed function, not on the "call_user_function".

So if $this->filter = 'trim' then the trim function is causing the error.  The $value needs to be converted to (string).  However, that might not be the solution for other filters.  So either check for a list of function that will not allow the $value to null and then check for a filter match and a null $value and converted the value to (string).

    If (in_array($this->$filter, $this->list_of_non_null_filters) && $value === null) {
        $model->$attribute = call_user_func($this->filter, (string)$value);
    } else {

Again, this might not be best handled here.

Please note that the description for the FilterValidator class offers a solution:

     * Many PHP functions qualify this signature (e.g. `trim()`).
     * If the callback function requires non-null argument (important since PHP 8.1)
     * remember to set [[skipOnEmpty]] to `true` otherwise you may trigger an error.